src/components/DisplayResult.vue

```diff
@@ -35,34 +39,45 @@ import { ref, onMounted, watch } from "vue";
 import { useRouter } from "@/router/injectRoute";
 
 const route = useRouter();
-const tag = ref("");
+const searchResult = ref("");
 const level = ref(5);
 
 onMounted(async () => {
   if (route) {
-    tag.value = route.currentRoute.value.query.tag as string;
-    level.value = Number(route.currentRoute.value.query.level);
-    await fetchData(tag.value, level.value);
+    const query = route.currentRoute.value.query;
+    searchResult.value = query.searchResult as string;
+    level.value = Number(query.level);
+    await fetchData(searchResult.value, level.value, query.type as string);
   } else {
-    tag.value = "undefined";
+    searchResult.value = "undefined";
   }
 });
-const fetchData = async (tag: string, level: number) => {
-  // TODO: also getFilteredResourcesByTitle here.
-
-  const results = await findData(
-    tag as string,
-    level as number,
-    getApisFromHost,
-  );
-
-  if (results) searchResults.value = results;
+const fetchData = async (searchResult: string, level: number, type: string) => {
+  if (!searchResult) return;
+
+  switch (type) {
+    case "tag":
+      searchResults.value = await findTagsData(
+        searchResult,
+        level,
+        getApisFromHost,
+      );
+      break;
+    case "title":
+      searchResults.value = await getFilteredResourcesByTitle(
+        searchResult,
+        searchFilterApi,
```

Comment:
LGTM! However, I'm wondering if we should include `level` as a query parameter, unless it's being used somewhere.